File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int res = 0;
        char[] chars = s.toCharArray();
        Set<Character> seen = new HashSet<>();
        int l = 0;
        for (int r = 0; r < s.length(); r++) {
            char c = chars[r];
            while (seen.contains(c)) {
                seen.remove(chars[l]);
                l++;
            }
            seen.add(c);
            res = Math.max(res, r - l + 1);
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        res = 1
        st = set(nums)
        for n in nums:
            curr = n
            count = 1
            while (curr + 1 in st):
                count += 1
                curr += 1
            res = max(res, count)
        return res